### PR TITLE
Add Github Deployment

### PR DESCRIPTION
**Pull request by jakobe**
This PR closes # (reference issue number here)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [X] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -46,6 +46,34 @@ jobs:
...
+        with:
+          route: POST /repos/:repository/deployments
+          repository: ${{ github.repository }}
+          ref: ${{ github.ref }}
+          environment: qa
+          transient_environment: true
+          auto_merge: false
+          required_contexts: '[]'
+          mediaType: '{"previews": ["flash", "ant-man"]}'
+        env:
```
Comment (bogedal) [inline .github/workflows/build.yml:62]: Burde man sætte GITHUB_TOKEN globalt?
ALtså noget ala 

`jobs:
  build:
    runs-on: ubuntu-latest
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
`

Comment (jakobe) [reply .github/workflows/build.yml:98]: Njaaa...

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -45,9 +45,69 @@ jobs:
...
+          echo "Removed uppercase - branchname: ${branchname}"
+          branchname="${branchname//[^a-zA-Z0-9]/-}"
+          echo "Removed illegal chars - branchname: ${branchname}"
+          branchname=$(tr -s '-' '-' <<< "$branchname")
+          echo "Removed duplicated dashes - branchname: ${branchname}"
+          branchname="${branchname%%-}"
+          echo "Removed trailing dash - final branchname: ${branchname}"
+          releasename="kby-${branchname}"
+          releasename=$(cut -c 1-53 <<< "$releasename")
+          echo "Releasename: ${releasename}"
```
Comment (jakobe) [inline .github/workflows/build.yml:70]: @bogedal I added the echo's as a way to document what's going on here - if you're not a bash shark (like I'm not 👶 ) then it was pretty hard to figure out what was going on (and therefore hard to spot why the branch name had the first character removed 🤔 )

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -45,9 +45,69 @@ jobs:
...
+          echo "Converted to lowercase - branchname: ${branchname}"
+          branchname="${branchname//[^a-zA-Z0-9]/-}"
+          echo "Replaced illegal chars - branchname: ${branchname}"
+          branchname=$(tr -s '-' '-' <<< "$branchname")
+          echo "Removed duplicated dashes - branchname: ${branchname}"
+          branchname="${branchname%%-}"
+          echo "Removed trailing dash - final branchname: ${branchname}"
+          releasename="kby-${branchname}"
+          releasename=$(cut -c 1-53 <<< "$releasename")
+          echo "Releasename: ${releasename}"
```
Comment (jakobe) [inline .github/workflows/build.yml:70]: @bogedal I added the echo's as a way to document what's going on here - if you're not a bash shark (like I'm not 👶 ) then it was pretty hard to figure out what was going on (and therefore hard to spot why the branch name had the first character removed 🤔 )

Comment (bogedal) [reply .github/workflows/build.yml:70]: Helps with readability, so super fine

File: .github/workflows/cleanup-branch-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+name: Clean up branch deployment
+on:
+  delete
+
+jobs:
+  delete_latest_deployment_for_branch:
+    name: Delete latest deployment for branch
+    if: github.event.ref_type == 'branch'
```
Comment (bogedal) [inline .github/workflows/cleanup-branch-deployment.yml:8]: Dont think this is needed as it can only be branch or tag

Comment (jakobe) [reply .github/workflows/cleanup-branch-deployment.yml:8]: Yep, but we don't want to trigger on tag deletion

**Review by bogedal — APPROVED** [review]
LGTM

**Pull request merged**